src/nsls2api/cli/facility.py

```diff
@@ -1,16 +1,66 @@
 import typer
+from rich.panel import Panel
+from rich.table import Table
 
-from nsls2api.services.facility_service import facility_cycles
+from nsls2api.api.models.facility_model import FacilityName
+from nsls2api.cli.utils.api import call_nsls2api_endpoint
+from nsls2api.cli.utils.cli_helpers import auto_help_if_no_command
+from nsls2api.cli.utils.console import console, error, info, warning
 
-app = typer.Typer()
+app = typer.Typer(invoke_without_command=True)
+
+
+@app.callback()
+@auto_help_if_no_command()
+def facility_callback(ctx: typer.Context):
+    pass  # No need to call anything manually
 
 
 @app.command()
-def view(facility: str):
-    print(f"Viewing Facility: {facility}")
+def view(facility: FacilityName):
+    info(f"Viewing Facility: [green]{facility}[/green]")
 
 
 @app.command()
-def cycles(facility: str):
-    cycle_list = facility_cycles(facility)
-    return cycle_list
+def cycles(facility: FacilityName):
+    """
+    Retrieve and display a list of cycles for a facility from the API.
+    """
+    endpoint = f"v1/facility/{facility}/cycles"
+    response = call_nsls2api_endpoint(endpoint, method="GET")
```

Comment:
Should perhaps the other CLI commands use `call_nsls2api_endpoint()` as done here, rather than handling the `httpx.Client()` directly?

Reply:
They should indeed.

Reply:
Have replaced these calls.